### DISCUSSION TITLE
Lesson 7: Receiving 'script failed: the path broadcast is not allowed to be accessed for read operations' when 'Interactions.s.sol' is ran.

### DISCUSSION THREAD

**Discussion by bhanupbalusu** [Cyfrin/foundry-full-course-cu#1302] [Q&A] [answered]
I am trying run the script 'Interactions.s.sol' using the following command:
forge script script/Interactions.s.sol --tc FundMeFoundryFund --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast -vvvvv

[27943] FundFundMe::run()
    ├─ [3925] DevOpsTools::get_most_recent_deployment("FundMe", 11155111 [1.115e7]) [delegatecall]
    │   ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   │   └─ ← the path broadcast is not allowed to be accessed for read operations
    │   └─ ← the path broadcast is not allowed to be accessed for read operations
    └─ ← the path broadcast is not allowed to be accessed for read operations

I have changed 'RELATIVE_BROADCAST_PATH = "./broadcast";' to 'RELATIVE_BROADCAST_PATH = "../broadcast";' as  DevOpsTools.sol is inside src folder, src folder and broadcast folder are in foundry-devops folder. Also changed the path from 'fs_permissions = [{ access = "read", path = "./broadcast" }]' to 'fs_permissions = [{ access = "read", path = "../broadcast" }]'. But I am still receiving same error.

Any help is appreciated. Thank you.

**Comment by APexBlockchainDevelopments** [comment, marked as answer]
I had to add this to my toml file. I had the same issue this seems to have fixed it. 
```
fs_permissions = [{ access = "read", path = "./broadcast" }]
```

**Reply by bhanupbalusu** [reply]
Thanks for the response [APexBlockchainDevelopments]. I have already had fs_permissions in place in toml file as mentioned earlier in my message, I also tweaked with the path parameter but still with no luck. Appreciate your response. Thanks.

**Reply by danail-vasilev** [reply]
I got the same issue.
After setting permissions, however, I got another issue (Note also that in the video these scripts are not run, just the integrations tests):

[⠢] Compiling...
No files changed, compilation skipped
Traces:
  [1139609] → new DevOpsTools@0x7FA9385bE102ac3EAc297483Dd6233D62b3e1496
    └─ ← 5750 bytes of code

  [226169] → new FundFundMe@0x34A1D3fff3958843C43aD80F30b94c510645C316
    └─ ← 1068 bytes of code

  [717500] FundFundMe::run()
    ├─ [711274] DevOpsTools::get_most_recent_deployment("FundMe", 31337 [3.133e4]) [delegatecall]
    │   ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   │   └─ ← [("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol", 1, true, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/11155111", 2, true, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/11155111/dry-run", 3, true, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337", 2, true, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-1706103797.json", 3, false, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-1706103929.json", 3, false, false), ("", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-latest.json", 3, false, false)]
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::toString(31337 [3.133e4]) [staticcall]
    │   │   └─ ← "31337"
    │   ├─ [0] VM::readFile("/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-latest.json") [staticcall]
    │   │   └─ ← <file>
    │   ├─ [0] VM::parseJsonUint("<stringified JSON>", ".timestamp") [staticcall]
    │   │   └─ ← 1706103929 [1.706e9]
    │   ├─ [0] console::log("Processing: ", "/Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-latest.json") [staticcall]
    │   │   └─ ← ()
    │   ├─ [0] VM::parseJson("<JSON file>", "$.transactions") [staticcall]
    │   │   └─ ← <encoded JSON value>
    │   ├─ [0] VM::keyExists("<JSON file>", "$.transactions[0].rpc") [staticcall]
    │   │   └─ ← false
    │   ├─ [0] console::log("Inspecting %s transactions", 2) [staticcall]
    │   │   └─ ← ()
    │   └─ ← 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
    ├─ [0] VM::startBroadcast()
    │   └─ ← ()
    └─ ← EvmError: Revert


Gas used: 738564

== Logs ==
  Processing:  /Users/user/Documents/repos/foundry-f23/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/31337/run-latest.json
  Inspecting 2 transactions
Error: 
script failed: <empty revert data>

**Reply by ProKing13** [reply]
Thanks bro it work for me

**Reply by TIP1** [reply]
same problem
did you resolved it?

**Comment by Blockchain-Oracle** [comment]
Your question is poorly formatted, you have low chance of getting an anwer. Try reformatting

**Comment by lamdan0901** [comment]
> I had to add this to my toml file. I had the same issue this seems to have fixed it.
> 
> ```
> fs_permissions = [{ access = "read", path = "./broadcast" }]
> ```

Luckily, this works for me!

**Reply by MartinPetrov12** [reply]
Worked for me as well!

**Comment by ProKing13** [comment]
> I had to add this to my toml file. I had the same issue this seems to have fixed it.
> 
> ```
> fs_permissions = [{ access = "read", path = "./broadcast" }]
> ```
